e2e/ansible/playbooks/resiliency/test-k8s-master-failure/k8s-master-failure.yml

```diff
@@ -0,0 +1,315 @@
+       shell: vagrant status kubemaster-01 | grep kubemaster-01 |  awk '{ $1="";print}' | cut -d '(' -f1
+       args:
+         chdir: "{{ansible_env.HOME}}/{{vagrant_path}}"
+         executable: /bin/bash
+       register: VDown
+       until: "'poweroff'  in VDown.stdout"
+
+
+     - name: Bring up the kubemaster.
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/vagrant-halt-up-task.yml"
+       vars:
+         vagrant_op: up 
+         nodename: kubemaster-01
+
+     - name: wait for 120 seconds.
+       wait_for:
+          timeout: 120
+
+     - name: 7a) Checking the status of kubemaster.
+       shell: vagrant status kubemaster-01 | grep kubemaster-01 |  awk '{ $1="";print}' | cut -d '(' -f1
+       args:
+         chdir: "{{ansible_env.HOME}}/{{vagrant_path}}"
+         executable: /bin/bash
+       register: Vup
+       until: "' running '  in Vup.stdout"
+
+
+     - name: Get jenkins  PV name
+       shell: source ~/.profile; kubectl get pv -n {{ namespace }}| grep {{namespace}}/jenkins-claim | awk '{print $1}'
+       args:
+         executable: /bin/bash
+       register: pv_name
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+ 
+ 
+     - name: Check the number of  available replicas in deployment
+       shell: kubectl get deploy -n {{ namespace }} | grep "{{ pv_name.stdout }}" | grep rep | awk '{print $2}'
+       args:
+         executable: /bin/bash
+       register: available_pods
+       until: "'3' in available_pods.stdout"
+       delay: 30
+       retries: 5
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       ignore_errors: true
+
+
+     - name: Confirm pod status is running after kubemaster up.
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+       vars:
+          ns: "{{ namespace }}"
+          app: jenkins
+
+     - name: 8a)  Check if the replica is rebuilded using mayactl commands
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/access-mode-check.yml"
+       vars:
+         ns: "{{ namespace }}"
+
+   
+ 
+     - name: 9)  Get a job configuration from previous build( get,create and build the job )
+       shell: >
+         source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }}
+         -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s
+         http://localhost:8080 get-job sample > /var/jenkins_home/template.xml";
+         source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }}
+         -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s 
+         http://localhost:8080 create-job sample2 < /var/jenkins_home/template.xml";
+         source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }}
+         -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s 
+         http://localhost:8080 build sample2 "
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
```

Comment:
Can this be optimized via script templates that can be pushed into the jenkins pod ? It can be invoked with desired args

Reply:
Yes, I created Build_config script file which is pushed inside the jenkins pod